### DISCUSSION TITLE
Constraining dependent node on LinearNodalConstraint

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#28945] [Q&A Modules: Solid mechanics]
Hi MOOSE team,

I have a question about the behaviour of LinearNodalConstraint. I have a problem where I am trying to constraint the _average_ displacement of the nodes on a boundary to 0. I have been trying to use a LinearNodalConstraint for this.

 I have a central dependent (or secondary as the LinearNodalConstraint calls it) node in the middle of the cylindrical boundary, with all the nodes on the boundary acting as the independent (Primary) nodes. The weightings for all of the nodes are equal to 1/(# of nodes). Therefore the displacement at the dependent node should be.

$$\Sigma u_i w_i = u_{dependent}$$

My question is, if I use a DirichletBC to prescribe the displacement of the dependent node to 0, will that then force,

$$\Sigma u_i w_i = 0$$

It feels wrong to prescribe the displacement of a _dependent_ node, that DOF seems like it would have to be removed twice from the system. MOOSE will happily solve the problem, but I don't know what it is doing behind the scenes and whether it is at all what I am actually trying to do. 

I have also tried using a lagrange multiplier approach to constraint the average displacement but the extra lagrange multiplier variables made preconditioning a pain. 

Any help or insight is greatly appreciated!

![Screenshot_20241028_133508](https://github.com/user-attachments/assets/86137009-638c-4ee5-aa29-404e08de8847)

**Comment by GiudGiud** [comment]
@lindsayad

**Comment by TheBEllis** [comment]
Since I posted this I have become aware that constraining the dependent node is a no go.

**Comment by megan-crocker** [comment]
Hi! I hope you don't mind me asking - did you manage to find an answer to this? Also, how did you create the secondary node? I'm trying to set up the exact same problem

Thanks :)

**Reply by GiudGiud** [reply]
I don't know about the first part. 

For the second part, we now have a mesh generator that can create a 1D line average of a nodeset. Maybe that could work here to create the secondary nodes. 

It's the PolyLineFollowingNodeSetGenerator
